eventuals/stream.h

```diff
@@ -169,46 +186,46 @@ struct _Stream final {
 
     template <typename... Args>
     void Start(Args&&... args) {
-      previous_ = Scheduler::Context::Get();
-
-      streamk_.stream_ = this;
-      streamk_.k_ = &k_;
-
       if constexpr (IsUndefined<Begin_>::value) {
-        streamk_.Begin(std::forward<Args>(args)...);
+        adaptor().Begin(std::forward<Args>(args)...);
       } else {
         if constexpr (!IsUndefined<Context_>::value && Interruptible_) {
           CHECK(handler_);
-          begin_(context_, streamk_, *handler_, std::forward<Args>(args)...);
+          begin_(context_, adaptor(), *handler_, std::forward<Args>(args)...);
         } else if constexpr (!IsUndefined<Context_>::value && !Interruptible_) {
-          begin_(context_, streamk_, std::forward<Args>(args)...);
+          begin_(context_, adaptor(), std::forward<Args>(args)...);
         } else if constexpr (IsUndefined<Context_>::value && Interruptible_) {
           CHECK(handler_);
-          begin_(streamk_, *handler_, std::forward<Args>(args)...);
+          begin_(adaptor(), *handler_, std::forward<Args>(args)...);
         } else {
-          begin_(streamk_, std::forward<Args>(args)...);
+          begin_(adaptor(), std::forward<Args>(args)...);
         }
       }
     }
 
     template <typename Error>
     void Fail(Error&& error) {
       if constexpr (IsUndefined<Fail_>::value) {
+        // Need to install 'previous_' and 'adaptor_' there.
+        // This condition would be executed only when 'fail' is
+        // not specified, what means that 'error' just propagates
+        // through and there is no need to check the 'error' type.
```

Comment:
```suggestion
        // We don't want to use 'adaptor_' here because we want to propagate what ever error is passed to us to 'k_' and 'adaptor_' expects specific errors but 
        // we do need to set 'previous_' in case a continuation 
        // calls 'Next()' or 'Done()'.
```